crates/shared/src/current_block.rs

```diff
@@ -124,13 +140,32 @@ where
     }
 }
 
+/// Only updates the current block if the new block number is strictly bigger than the current one.
+/// Emits error logs when the new block number differs by more than `MAX_REORG_BLOCK_COUNT` from
+/// the current block number.
+fn block_number_increased(current_block: &AtomicU64, new_block: u64) -> bool {
+    let current_block = current_block.fetch_max(new_block, Ordering::SeqCst);
+    if new_block > current_block + MAX_REORG_BLOCK_COUNT {
+        tracing::error!(new_block, "received a block that is suspiciously far ahead");
+        true
+    } else if new_block > current_block {
+        tracing::debug!(new_block, "received a block that is reasonably far ahead");
+        true
+    } else if new_block > current_block.saturating_sub(MAX_REORG_BLOCK_COUNT) {
+        tracing::warn!(new_block, "received a slightly outdated block");
+        false
```

Comment:
Should this really be a warning? In PoW this is normal behavior and is not warn-worthy. Maybe in PoS it shouldn't happen?

Reply:
I guess that depends on what our model of "normal" refers to. I.e. for PoW it would be normal but for how we want our load balancer to work it wouldn't be. I don't have a strong preference, though.

Reply:
I think we configured our load balancer to allow small block propagation delays (cc @ahhda) - if I remember correctly we had said that 2 blocks behind is too much, but 1 block behind is OK.

So, in many cases it would be "normal", in some "abnormal" and merit a warning.

The improvement I could think of would be to increment metrics when we see these different conditions (something like a histogram with different buckets for different block differences). That being said, I don't want to complicate this too much, and what we have in this PR is already a HUGE improvement that I am happy with merging.

Reply:
I didn't think about using a histogram for that 🤦‍♂️ . I preferred that idea over the `if-else` cascade. Note however, that we also have to hardcode the buckets of the histogram so we are not completely flexible either.
I decided to use following buckets because they seemed the most reasonable: `0, 1, 2, 4, 8, 25`
Also since a single histogram with positive and negative values turned out to be tricky to use in Grafana while Dusan worked on the access lists metrics I decided to follow suit and have one histogram for positive and one for negative values, too. (0 gets booked in the negative histogram).